### PR TITLE
Do not show download bars in non interactive runs

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

As discussed in the Refinement session, async tqdm bars can make some [CI logs too long](https://github.com/conda/constructor/actions/runs/5799411468/job/15719353263#step:3:286) and makes reading these harder. In this PR I check whether `sys.stdout` is a TTY (like the `Spinner()` class does, and decide whether to create progress bars or not. Some extra care has been taking to adjust the output a bit so it doesn't look like it's hung.

Example output: [long-log.txt](https://github.com/conda/conda/files/12294913/long-log.txt)

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

cc @dholth

### CODE REVIEW

**Comment by dholth** [comment]
Looks OK. Could use better test coverage. Would it be a good idea to configure what's printed at each step through variables and remove some of the if: statements?

File: conda/common/io.py [hunk 1]
```diff
@@ -533,6 +543,12 @@ def _tqdm(*args, **kwargs):
 
         return tqdm(*args, **kwargs)
 
+    @staticmethod
+    @lru_cache(maxsize=1)
+    def interactive():
+        """Return True if we're running in a terminal."""
+        return hasattr(sys.stdout, "isatty") and sys.stdout.isatty()
```
Comment (jezdez) [inline conda/common/io.py:550]: I'm not a fan of static methods, could we just move this into a separate module level function or even just a constant?

```
is_interactive = hasattr(sys.stdout, "isatty") and sys.stdout.isatty()
```

**Review by jezdez — CHANGES_REQUESTED** [review]
Let's move the check into a constant instead, no need to have an LRU cache if we have it cached in the module

File: conda/common/io.py (hunk 1 above)
Comment (jaimergp) [reply conda/common/io.py]: Done!

File: conda/common/io.py [hunk 2]
```diff
@@ -473,33 +473,40 @@ def __init__(
...
+                        leave=leave,
+                    )
+                except OSError as e:
+                    if e.errno in (EPIPE, ESHUTDOWN):
+                        self.enabled = False
+                    else:
+                        raise
+            else:
+                self.pbar = None
+                sys.stdout.write("%s ...working..." % description)
```
[the remark this answers is not in the document]
Comment (jaimergp) [reply conda/common/io.py:494]: I can change everything in the module to f-strings if you want, but that'll create a bigger diff. Up to you!

[the remark this answers is not in the document]
Comment (jezdez) [reply conda/common/io.py:494]: Nah, just wanted to leave my comment here for posterity :)

**Review by jezdez — APPROVED** [review]

**Pull request merged**